layers/meta-opentrons/recipes-robot/opentrons-robot-app/files/opentrons-loading.sh

```diff
@@ -1,16 +1,54 @@
 #!/bin/sh
 
+# This script is responsible for setting the splash screen when the
+# system boots up. We export the /var/lib/opentrons-system-server/system.env
+# environment file to determine if the system is in OEM Mode, we then
+# set the splash screen accordingly.
+
 # Get the brightness set as early as possibly to prevent flickers
 echo 1 > /sys/class/backlight/backlight/brightness
 
-# Absolute path to the video location
-VIDEO=$1
-
 # Graceful cleanup upon CTRL-C
 trap "gstd-client pipeline_delete p; exit" SIGHUP SIGINT SIGTERM
 
+# check the environment file exists, otherside just default to opentrons loading screen
+system_env_file="/var/lib/opentrons-system-server/system.env"
+if [ -f $system_env_file ]; then
+	echo "Found system environment file: ${system_env_file}"
+	export $(grep -v '^#' $system_env_file | xargs)
+fi
+
+oem_mode_enabled=$OT_SYSTEM_SERVER_oem_mode_enabled
+oem_mode_splash_custom=$OT_SYSTEM_SERVER_oem_mode_splash_custom
+oem_mode_splash_default="/usr/share/opentrons/oem_mode_default.png"
+opentrons_default_splash="/usr/share/opentrons/loading.mp4"
+splash_screen_path="${opentrons_default_splash}"
+if [[ $oem_mode_enabled == 1 ]]; then
+	echo "OEM Mode is Enabled"
+	if [ -f $oem_mode_splash_custom ]; then
+		echo "Custom OEM file is set: ${oem_mode_splash_custom}"
+		splash_screen_path=$oem_mode_splash_custom
+	else
+		echo "Default OEM file is set: ${oem_mode_splash_default}"
+		splash_screen_path=$oem_mode_splash_default
+	fi
+fi
+
+# Make sure the file exists
+if [ ! -f $splash_screen_path ]; then
+	echo "ERROR: Splash screen file not found: ${splash_screen_path}"
+	exit 1
```

Comment:
this will make systemd consider this unit failed which might impact the app starting at all. may want to note the problem and have a fallback, or in any case exit 0 so it's still a success